### PR TITLE
fix: explicitly use surface TKE in mixing length computation

### PR DESCRIPTION
**Pull request by haakon-e**
This PR fixes an issue where you in certain circumstances could get `TKE=0.0` at the surface, which resulted in a `ZeroDivisionError` in the computation of the near-surface mixing length
https://github.com/CliMA/SCAMPy/blob/3512b8c56030e7a892aef37ecef2896da34229ed/Turbulence_PrognosticTKE.pyx#L792-L796
This issue is related to the solution of the TKE equation, which, if it is negative, gets clipped to zero
https://github.com/CliMA/SCAMPy/blob/3512b8c56030e7a892aef37ecef2896da34229ed/Turbulence_PrognosticTKE.pyx#L2344-L2352
In fact, the surface TKE should be computed by `get_surface_tke`, As such, we propose to use the `get_surface_tke` function directly in the mixing length closure.

some other stylistic changes and updates to docs is also included.

### CODE REVIEW

**Review by yairchn — APPROVED** [review]
for some reason we don't have the 
'namelist['output']['output_root'],' replacing 'namelist['output']['output_root']+'
in line 42 in test/commun.py. 
The current version breaks for test on my machine 
maybe we can add this ?

I would also set line 62 in test/commun.py to 
cmd = "rm -r Tests./Output.*"

After fixing this the test run and results look good.

File: Turbulence_PrognosticTKE.pyx [hunk 1]
```diff
@@ -875,14 +889,27 @@ cdef class EDMF_PrognosticTKE(ParameterizationBase):
                 self.ml_ratio[k] = self.mixing_length[k]/l[int(self.mls[k])]
 
         elif (self.mixing_scheme == 'sbtd_eq'):
+            # Ref: Lopez-Gomez et al. (2020)
```
[the remark this answers is not in the document]
Comment (haakon-e) [reply Turbulence_PrognosticTKE.pyx:892]: oh, it's true for all mixing length schemes? Okay. We can add the reference add the top or something.

[the remark this answers is not in the document]
Comment (ilopezgp) [reply Turbulence_PrognosticTKE.pyx:892]: yes. You can add it at the top of the mixing length function.

**Review by ilopezgp — APPROVED** [review]
LGTM

**Comment by yairchn** [comment]
I think this change is also needed in case you run a test, it breaks (so you haven't remove the files) and than you run it again. In this case the old files should be remove in the beginning of the simulation or you will have a "Restart" run. 
in line 62 in test/commun.py set 
cmd = "rm -r Tests./Output.*"

**Pull request merged**